.github/workflows/pull-request-check-rust-api.yaml

```diff
@@ -0,0 +1,105 @@
+name: Build and Test the Rust API
+on:
+  push:
+    branches: [ develop ]
+  pull_request:
+    branches: [ develop ]
+env:
+  default_build_dir: "build/"
+  default_solver: "minisat2"
+
+# For now, we support two jobs: A Linux and a MacOS based one.
+# Both of the jobs use CMake, as we only support building the Rust
+# API with CMake (due to dependencies on CMake plugins).
+jobs:
+  check-ubuntu-22_04-cmake-clang-rust:
+    runs-on: ubuntu-22.04
+    env:
+      CC: "ccache /usr/bin/clang-13"
+      CXX: "ccache /usr/bin/clang++-13"
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Fetch dependencies
+        env:
+          # This is needed in addition to -yq to prevent apt-get from asking for
+          # user input
+          DEBIAN_FRONTEND: noninteractive
+        run: |
+          sudo apt-get update
+          sudo apt-get install --no-install-recommends -yq clang-13 clang++-13 gdb maven jq flex bison libxml2-utils cpanminus ccache z3
+          make -C src minisat2-download
+          cpanm Thread::Pool::Simple
+      - name: Confirm z3 solver is available and log the version installed
+        run: z3 --version
+      - name: Log cargo/rust version
+        run: cargo --version
+      - name: Prepare ccache
+        uses: actions/cache@v3
+        with:
+          path: .ccache
+          key: ${{ runner.os }}-22.04-cmake-clang-${{ github.ref }}-${{ github.sha }}-PR
+          restore-keys: |
+            ${{ runner.os }}-22.04-cmake-clang-${{ github.ref }}
+            ${{ runner.os }}-22.04-cmake-clang
+      - name: ccache environment
+        run: |
+          echo "CCACHE_BASEDIR=$PWD" >> $GITHUB_ENV
+          echo "CCACHE_DIR=$PWD/.ccache" >> $GITHUB_ENV
+      - name: Zero ccache stats and limit in size
+        run: ccache -z --max-size=500M
+      - name: Configure using CMake
+        run: cmake -S. -B${{env.default_build_dir}} -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_C_COMPILER=/usr/bin/clang-13 -DCMAKE_CXX_COMPILER=/usr/bin/clang++-13 -DWITH_RUST_API=ON
```

Comment:
Could you please explain _why_ you need PIC? The commit message, unfortunately, lacks a body (which I would have hoped to explain this move).

Reply:
The simple answer is that CI failed without it (reproducibly on CI, claims that it needs everything recompiled with `-fPIE`.

Worth mentioning that Enrico couldn't reproduce locally on his machine, but given that this got the build to complete and do the actual Rust test run, I'm happy to leave this in for this particular configuration (doesn't seem to do any harm).

Reply:
No harm is fine, but the reasons should be documented in some form.

Reply:
This has now been documented in the form of a documentation comment in the same action script, explaining the need for the flag.